### DISCUSSION TITLE
Does React XR work on iOS/Android?

### DISCUSSION THREAD

**Discussion by Acr515** [pmndrs/xr#224] [Q&A] [answered]
I haven't found any documentation for or against support in mobile browsers, but I wanted to make sure I checked. I'm just getting started with this library, but every mobile device I've tried with the `ARButton` shows "AR Unsupported." `VRButton` behaves as intended on my Oculus, but it, too, shows "VR Unsupported" on other devices (I assumed it would do this either way; I'm more focused on AR functionality). I should also note that I'm on an HTTPS connection each time I access the React application.

Does React XR work in iOS and Android browsers, or am I just implementing it incorrectly?

**Comment by CodyJasonBennett** [comment, marked as answer]
You can find general device support in https://caniuse.com/webxr, but WebXR is not implemented in iOS Safari. It does work on most recent Androids I've tested of the Pixel lineup for AR modes, but Android is too diverse with old, ancient hardware to ensure compat. MacOS does not implement OpenXR either, although I've seen independent implementations, unfortunately they've locked down their ecosystem so there's no such thing as an external contribution -- they don't want Open/WebXR.

If you want good mobile support, I'd look into something that implements its own computer vision like ARjs, or a corporate backed implementation like 8thwall. Otherwise, you could go native and connect to ARCore/ARKit. I've been looking into writing a polyfill, but it is quite the undertaking in both work and timeline, so definitely find a corporate-backed solution if your company permits.

**Reply by Acr515** [reply]
Thanks for the info! I'll read up on some alternatives.

**Reply by kordou** [reply]
did you found any alternatives that work for both ios and android ?

**Reply by Acr515** [reply]
@kordou Haven't revisited this project in quite a while, but I had bookmarked [this library built upon AR.js](https://github.com/artcom/react-three-arjs) that seems promising, just haven't tried using it yet.
